### PR TITLE
Fix Default Lovelace Panel Title

### PR DESCRIPTION
**Pull request by timmo001**
## Proposed change

Fixes the missing panel title on the default lovelace view. I did get this working in #5220 so I'm not sure what's changed since. 

_Could be from the removal of the states ui but I'm unsure_

Screenshot_20200321_111257

## Type of change

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

### CODE REVIEW

File: src/data/panel.ts [hunk 1]
```diff
@@ -9,27 +9,26 @@ export const getDefaultPanelUrlPath = () =>
 export const getDefaultPanel = (panels: HomeAssistant["panels"]) =>
   panels[localStorage.defaultPage] || panels[DEFAULT_PANEL];
 
-export const getPanelTitle = (hass: HomeAssistant): string | undefined => {
+export const getPanelTitle = (hass: HomeAssistant): string | null => {
   if (!hass.panels) {
-    return;
+    return null;
```
Comment (bramkragten) [inline src/data/panel.ts]: Why return `null` instead of `undefined`?

File: src/data/panel.ts [hunk 2]
```diff
@@ -9,27 +9,26 @@ export const getDefaultPanelUrlPath = () =>
...
     return hass.localize("panel.profile");
   }
 
-  return (
-    hass.localize(`panel.${panel.title}`) ||
-    panel.title ||
-    // default panel
-    (hass.panels[localStorage.defaultPage] || hass.panels[DEFAULT_PANEL]).title!
-  );
+  return hass.localize(`panel.${panel.title}`) || panel.title;
```
Comment (bramkragten) [inline src/data/panel.ts]: Now other Lovelace panels don't have the right title

File: src/data/panel.ts (hunk 1 above)
Comment (McGiverGim) [reply src/data/panel.ts]: I'm not the developer of this but I was curious about this too and for example Sonar suggests in its rules to use always null.

Looking more articles in the web "The main difference is that undefined represents the value of a variable that wasn’t yet initialized, while null represents an intentional absence of an object."

To me it seems intentional here that he wants to return "nothing", so it seems ok to me return null.

Comment (timmo001) [reply src/data/panel.ts]: `panel.title` has the type `string | null` which caused the ts error.

I've added additional logic to revert this change

File: src/data/panel.ts [hunk 3]
```diff
@@ -22,14 +22,19 @@ export const getPanelTitle = (hass: HomeAssistant): string | undefined => {
...
-    // default panel
-    (hass.panels[localStorage.defaultPage] || hass.panels[DEFAULT_PANEL]).title!
-  );
+  const title = hass.localize(`panel.${panel.title}`) || panel.title;
+
+  if (!title) {
+    return;
+  }
+
+  return title;
```
Comment (bramkragten) [inline src/data/panel.ts]:
```suggestion
  return hass.localize(`panel.${panel.title}`) || panel.title || undefined;
```

File: src/data/panel.ts [hunk 4]
```diff
@@ -22,14 +22,19 @@ export const getPanelTitle = (hass: HomeAssistant): string | undefined => {
...
+
   if (panel.url_path === "profile") {
     return hass.localize("panel.profile");
   }
 
-  return (
-    hass.localize(`panel.${panel.title}`) ||
-    panel.title ||
-    // default panel
-    (hass.panels[localStorage.defaultPage] || hass.panels[DEFAULT_PANEL]).title!
```
Comment (bramkragten) [inline src/data/panel.ts]: Why is this removed? Isn't it needed anymore? I would think it is still needed when `lovelace` is not the default panel?

**Review by bramkragten — APPROVED** [review]

File: src/data/panel.ts [hunk 5]
```diff
@@ -22,14 +22,19 @@ export const getPanelTitle = (hass: HomeAssistant): string | undefined => {
...
-  );
+  const title = hass.localize(`panel.${panel.title}`) || panel.title;
+
+  if (!title) {
+    return;
+  }
```
Comment (bramkragten) [inline src/data/panel.ts]: This can now be removed?

Comment (timmo001) [reply src/data/panel.ts]: Yep
